### PR TITLE
MAISTRA-2293 chart updates for MeshFederation resource support

### PR DESCRIPTION
**Pull request by rcernich**

### CODE REVIEW

File: resources/helm/v2.1/istio-control/istio-discovery/templates/deployment.yaml [hunk 1]
```diff
@@ -93,6 +93,8 @@ spec:
             protocol: TCP
           - containerPort: 15053
             protocol: TCP
+          - containerPort: 8188
```
Comment (dgn) [inline resources/helm/v2.1/istio-control/istio-discovery/templates/deployment.yaml:99]: thanks! was about to file a separate issue for this

File: resources/helm/v2.1/gateways/istio-ingress/templates/meshexpansion.yaml [hunk 2]
```diff
@@ -37,22 +37,22 @@ metadata:
     release: {{ .Release.Name }}
 spec:
   hosts:
-  - istiod.{{ .Release.Namespace }}.svc.{{ .Values.global.proxy.clusterDomain }}
+  - istiod-{{ .Values.revision | default "default" }}.{{ .Release.Namespace }}.svc.{{ .Values.global.proxy.clusterDomain }}
```
Comment (dgn) [inline resources/helm/v2.1/gateways/istio-ingress/templates/meshexpansion.yaml:40]: this is correct- but do we need the meshexpansion stuff? we don't support it, right?

Comment (rcernich) [reply resources/helm/v2.1/gateways/istio-ingress/templates/meshexpansion.yaml:40]: We don't, but my original tests were using MeshNetworks, so I had to get this working for that.  Didn't want to just throw the changes away.

Comment (dgn) [reply resources/helm/v2.1/gateways/istio-ingress/templates/meshexpansion.yaml:40]: gotcha

**Comment by rcernich** [comment]
@luksa, @dgn, do you guys mind reviewing this?  This is the operator side of the MeshFederation support.  Installs may be broken until this is merged, as the upstream changes in maistra/istio have already been merged.

File: resources/helm/v2.1/istio-init/files/servicemeshextensions.maistra.io.crd.yaml [hunk 3]
```diff
@@ -1,50 +1,53 @@
...
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   labels:
     maistra-version: "2.1.0"
-    chart: istio
-    release: istio
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.0
+    controller-gen.kubebuilder.io/version: v0.2.4
```
Comment (jwendell) [inline resources/helm/v2.1/istio-init/files/servicemeshextensions.maistra.io.crd.yaml:7]: Hopefully with the move to maistra/api we are having only one version. Currently operator is 0.4 and istio is 0.2.4

**Review by jwendell — APPROVED** [review]

File: resources/helm/v2.1/istio-init/files/servicemeshextensions.maistra.io.crd.yaml [hunk 4]
```diff
@@ -1,50 +1,53 @@
...
 spec:
   group: maistra.io
   names:
-    categories:
-    - maistra-io
     kind: ServiceMeshExtension
     listKind: ServiceMeshExtensionList
     plural: servicemeshextensions
-    shortNames:
-    - sme
```
Comment (dgn) [inline resources/helm/v2.1/istio-init/files/servicemeshextensions.maistra.io.crd.yaml:22]: losing shortname and category :slightly_frowning_face: we definitely have to move this to maistra/api and use code-gen 0.4.0+

**Review by dgn — APPROVED** [review]

**Pull request merged**